Board refactor, sensors and NanoPi R2S, RPi 3B + CM4 support

troglobit opened the pull request:
## Description

**New Hardware Support:**
- NanoPi R2S (Tier 2, added to default aarch64 build)
- Raspberry Pi 3B (BCM2837)
- Raspberry Pi CM4 variants, Waveshare IoT Router Board Mini and NVME NAS box

**Board Infrastructure:**
- Relocated `src/board/*` and `package/board/*` to unified `board/` directory
- Renamed `raspberry-pi-4` → `raspberrypi-rpi64`
- Consolidated SD card image generation into `utils/mkimage.sh`
- Extended `/var` partition auto-expansion to all MMC devices
- Renamed `mt7986_sd_boot_defconfig` → `bpi_r3_boot_defconfig`

**USB & Drivers:**
- Fixed duplicate USB port names in discovery
- Made USB subsystem built-in
- Added common USB-to-Ethernet drivers

## Hardware Monitoring & Sensors

<img align="right" width="283" height="965" alt="show-sys" src="https://github.com/user-attachments/assets/629d7cbc-9fa7-46fe-8e13-93c5330fd48c" />

**YANG Models:**
- `infix-hardware@2025-10-18`: Added `wifi` identity, enabled parent/child hierarchy
- `infix-system@2025-10-18`: Added resource usage and service monitoring

**Sensor Support:**
- Complete hwmon sensor discovery: temp, fan, voltage, current, power
- Hierarchical display with parent/child relationships for multi-sensor devices
- WiFi PHY identification with band detection (2.4/5/6 GHz)
- Human-readable descriptions from hwmon labels

**CLI:**
- New `show system`: hostname, uptime, load, CPU temp, fan, memory, disk
- Enhanced `show hardware`: hierarchical sensors, improved formatting, color-coded temps
- Fixed WiFi PHY to interface mapping (phy0→wifi0, phy1→wifi1)

## Configuration
- Updated BPi-R3 factory-config with full network setup
- Synced NanoPi R2S factory-config (bridge, DHCP, firewall, crypto types)
- Fixed firewall zone description type (was int, now string)

## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [x] Bugfix
  - [ ] Regression tests
  - [x] ChangeLog updates (for next release)
- [x] Feature
  - [ ] YANG model change => revision updated?
  - [ ] Regression tests added?
  - [x] ChangeLog updates (for next release)
  - [ ] Documentation added?
- [ ] Test changes
  - [ ] Checked in changed Readme.adoc (make test-spec)
  - [ ] Added new test to group Readme.adoc and yaml file
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [ ] Build related changes
- [ ] Documentation content changes
  - [ ] ChangeLog updated (for major changes)
- [ ] Other (please describe):

wkz commented on board/aarch64/friendlyarm-nanopi-r2s/rootfs/usr/share/product/friendlyarm,nanopi-r2s/etc/factory-config.cfg:
```diff
@@ -168,7 +150,17 @@
   },
   "ietf-system:system": {
-    "hostname": "infix-00-00-00",
+    "hostname": "r2s",
```
```suggestion
    "hostname": "r2s-%m",
```
Same here

troglobit replied:
OK.

wkz commented on board/ix-board.mk:
```diff
@@ -4,7 +4,12 @@ $(2)_VERSION = ix-board
+ifndef $(2)_BOARD_PATH
+$(2)_BOARD_PATH = $(3)
+endif
+$(2)_SITE = $$(BR2_EXTERNAL_INFIX_PATH)/board/$$($(2)_BOARD_PATH)
```
```suggestion
$(2)_SITE ?= $$(BR2_EXTERNAL_INFIX_PATH)/board/$(3)
```
And then if a board needs a custom path (e.g., because it is in a different external, then it can set `SITE` before calling the macro. Would that break anything?

wkz commented:
One more thing: have a look at the matching stuff I added to [`has-quirk`](https://github.com/kernelkit/infix/blob/main/board/common/rootfs/usr/libexec/infix/has-quirk). With that, you can match on things like the driver name, rather than just interface names.

troglobit commented on board/aarch64/friendlyarm-nanopi-r2s/rootfs/usr/share/product/friendlyarm,nanopi-r2s/etc/product/init.d/S01-trigger-udev.sh:
These triggers sometimes clash which is why we've made them board specific.  Unfortunately udev starts before the `init.d/` scripts run and install these udev rules.  We've spent a good while thinking about how to fix this, but not yet come up with any good ideas.

troglobit commented on board/common/post-build.sh:
```diff
@@ -101,25 +98,32 @@ fi
+		"$script"
+	fi
 done
+
```
OK, removing.

troglobit commented on board/aarch64/bananapi-bpi-r3/rootfs/usr/share/product/bananapi,bpi-r3/etc/factory-config.cfg:
```diff
@@ -0,0 +1,349 @@
+    }
+  },
+  "ietf-system:system": {
+    "hostname": "bpi-r3",
```
OK, fixing.

troglobit commented:
> One more thing: have a look at the matching stuff I added to [`has-quirk`](https://github.com/kernelkit/infix/blob/main/board/common/rootfs/usr/libexec/infix/has-quirk). With that, you can match on things like the driver name, rather than just interface names.

Thanks for that, didn't even know we had that type of matching! 

I've addressed almost all of your comments, only one was squashed into an existing commit (the big board refactor), the others are follow-up commits starting with `board/aarch64:`.  The one comment I cannot address is the udev rules chicken-and-egg situation, which I believe would require a significant retake on the Infix bootstrap process.

mattiaswal commented on src/confd/yang/confd/infix-hardware.yang:
```diff
@@ -34,25 +38,17 @@ module infix-hardware {
     description "A two-letter country code.";
   }
 
-  identity hardware-class {
```
This change was intentional, since in configuration you will get **all** hardware classes from iana. But we only support USB.

mattiaswal replied:
Then you need to add an upgrade script and step up the version for confd (for USB configuration)

troglobit replied:
Seems we don't actually need to.  Here's the old syntax, from an RPi4:

```json
  /* From static factory-config.cfg, unchanged since the start of time */
  "ietf-hardware:hardware": {
    "component": [
      {
        "name": "USB",
        "class": "infix-hardware:usb",
        "state": {
          "admin-state": "unlocked"
        }
      }
    ]
  },
```

vs new syntax:

```json
  /* From /run/confd/factory-config.gen */
  "ietf-hardware:hardware": {
    "component": [
      {
        "class": "infix-hardware:usb",
        "name": "USB1",
        "state": {
          "admin-state": "unlocked"
        }
      },
      {
        "class": "infix-hardware:usb",
        "name": "USB2",
        "state": {
          "admin-state": "unlocked"
        }
      },
      {
        "class": "infix-hardware:usb",
        "name": "USB3",
        "state": {
          "admin-state": "unlocked"
        }
      }
    ]
  },
```

The change in naming and amount of ports has changed because of fixing #315 and #1065 which is unfortunate but does not break the YANG model/config.

mattiaswal reviewed (approved):
Great work! :100:
